lib/puppet/type/cisco_bgp_af.rb

```diff
@@ -459,6 +469,112 @@ def insync?(is)
     end
   end # property :redistribute
 
+  newproperty(:route_target_both_auto) do
+    desc "Enable/Disable route target both auto only for IBGP.
+          Valid values are true, false, or 'default'"
+
+    newvalues(:true, :false, :default)
+  end # property route_target_both_auto
+
+  newproperty(:route_target_both_auto_evpn) do
+    desc "Enable/Disable route target both auto evpn only for IBGP.
+          Valid values are true, false, or 'default'"
+
+    newvalues(:true, :false, :default)
+  end # property route_target_both_auto_evpn
+
+  newproperty(:route_target_import, array_matching: :all) do
+    desc "Set route target import extended communities.
+          Valid values are Array, String, keyword 'default'"
+
+    match_error = 'must be specified in AS:nn or IPv4:nn notation'
+
+    validate do |community|
+      community.split.each do |value|
+        fail "Confederation peer value '#{value}' #{match_error}" unless
+          /^(?:\d+\.\d+\.\d+\.)?\d+.\d+$/.match(value) ||
```

Comment:
This regex isn't quite right. The `.` is a match any single character so it will match on `111`, `1X1`, etc. Use this instead:

`/^(?:\d+\.\d+\.\d+\.)?\d+:\d+$/`